client/components/Editor/commands/rtl.ts

```diff
@@ -0,0 +1,63 @@
+import { Node } from 'prosemirror-model';
+import { EditorState } from 'prosemirror-state';
+
+import { Dispatch } from './types';
+import { createCommandSpec } from './util';
+
+type NodePos = { node: Node; pos: number };
+const directionAttr = 'rtl';
+
+// or type is attr?
+const getDirectionValue = (direction: string, nodes: NodePos[]) => {
+	return [...nodes][0].node.attrs.rtl === direction ? null : direction;
+};
+
+// what nodes able to be rtl, not sure, how to check?
+
+const isRtlTarget = (node: Node) => {
+	const { spec } = node.type;
+	return spec.attrs && directionAttr in spec.attrs;
+};
+
+// nodes could be in ltr or rtl
+const getDirectionNodes = (state: EditorState): NodePos[] => {
+	const { selection, doc } = state;
+	const alignableNodes: NodePos[] = [];
+	selection.ranges.forEach(({ $from, $to }) => {
+		doc.nodesBetween($from.pos, $to.pos, (node: Node, pos: number) => {
+			if (isRtlTarget(node)) {
+				alignableNodes.push({ pos, node });
+			}
+		});
+	});
+
+	return alignableNodes;
+};
+
+const orientNodes = (
+	nodes: NodePos[],
+	state: EditorState,
+	dispatch: Dispatch,
+	attr: string | null,
+) => {
+	const { tr } = state;
+	nodes.forEach(({ pos, node }) =>
+		tr.setNodeMarkup(pos, undefined, { ...node.attrs, [directionAttr]: attr }),
+	);
+	dispatch(tr);
+};
+
+const createDirectionCommandSpec = (direction: string) => {
+	return createCommandSpec((dispatch: Dispatch, state: EditorState) => {
+		const nodes = getDirectionNodes(state);
+		const canRun = true; // not sure ?
+		const directionAttrValue = getDirectionValue(direction, nodes);
+		return {
+			isActive: false,
+			canRun,
+			run: () => orientNodes(nodes, state, dispatch, directionAttrValue),
+		};
+	});
+};
+
+export const rtlToggle = createDirectionCommandSpec('true');
```

Comment:
I want to mention that there are be better ways to represent the RTL direction than the string `'true'`. For instance, you could store the RTL-ness using `true | null`, and then write:

```ts
{ 'data-rtl': String(node.attrs.rtl) } 
```

when actually specifying DOM nodes in `base.ts`. It feels cleaner to use the least powerful data type required (in this case, a boolean).

Regardless, there should be a TypeScript type in this function somewhere that helps a reader understand what values the `rtl` attribute can take on.